queries/vm_backup.kusto

```diff
@@ -0,0 +1,24 @@
+let VMs = MonitoringResources_CL 
+| where TimeGenerated > ago(13h)
+| where type_s == "microsoft.compute/virtualmachines"
+| summarize take_anyif(TimeGenerated, isnotempty(properties_extended_instanceView_osName_s) ) by name_s, tags_managedby_s, properties_storageProfile_osDisk_osType_s, ResourceId
+| project vmResourceID = ResourceId, name = tolower(name_s), managedby = tags_managedby_s, osType = properties_storageProfile_osDisk_osType_s;
+AddonAzureBackupJobs
+| where JobStatus == "Failed"
+| where JobOperation =="Backup"
+| extend vm = extract("(iaasvmcontainerv2;.+?;)(.+)", 2, BackupItemUniqueId)
+| extend vault = extract("(microsoft.recoveryservices/vaults/)(.+)$", 2, tostring(_ResourceId))
+| project TimeGenerated, JobFailureCode, vm, vault, _ResourceId, JobOperationSubType, JobStartDateTime, BackupItemUniqueId, AdHocOrScheduledJob
+| join kind = leftouter(VMs) on $left.vm == $right.name
+| where managedby == "q.beyond"
```

Comment:
Why do a left outer join to definitely filter everything that got no match in the right table?
```suggestion
| join VMs on $left.vm == $right.name
| where managedby == "q.beyond"
```

Reply:
What do you mean? This join will include failed jobs even if no VM was found in the MonitorinResources_CL. When doing just an inner join we might miss failed jobs if there is no VM info, no? I guess in that case the query pretty much breaks either way with missing columns for the _ResourceId but better than not reporting the failed job at all.

Reply:
Nope it wont. The where clause afterwards makes sure, that I can't keep anything that had no match in the right table. As `managedby` can only be `q.beyond` if there was a match in the right table anyway.

Reply:
We decided this is fine for this release :)
A future update will include redundancy by using a fallback CI but this requires some additional work internally first.